### PR TITLE
Migrate Call List Entry model

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

Merge over Call List stuff.

We are almost back to being able to load the call list - still need notes, but getting ever closer!

To test:

```
git checkout main && rails db:drop db:create db:migrate db:seed
git checkout pg-calllist && rails db:migrate migrate_to_pg:clinic_user_patient
rails c
CallListEntry.count # => 6
```

This pull request makes the following changes:
* migrate call list entry model

no view changes

It relates to the following issue #s: 
* Bumps #2072

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: app/models/call_list_entry.rb [hunk 1]
```diff
@@ -1,25 +1,10 @@
...
-  field :line, type: String
-
   # Validations
   validates :order_key, :line, presence: true
   validates :patient, uniqueness: { scope: :user }
-
-  # Indices
-  index line: 1
-
-  def self.destroy_orphaned_entries
```
Comment (lomky) [inline app/models/call_list_entry.rb:21]: still one call to this hanging out in `lib`

```
Sun May 02 20:22:12 [pg-calllist] ~/repos/dcaf $ gg -C2 destroy_orphaned_entries
lib/tasks/nightly_cleanup.rake-  puts "#{Time.now} - removed old sessions"
lib/tasks/nightly_cleanup.rake-
lib/tasks/nightly_cleanup.rake:  CallListEntry.destroy_orphaned_entries
lib/tasks/nightly_cleanup.rake-  puts "#{Time.now} - removed orphaned call list entries"
lib/tasks/nightly_cleanup.rake-
```

File: app/models/user.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 class User < ApplicationRecord
   # Concerns
   include PaperTrailable
-  # include CallListable # Soon
```
Comment (lomky) [inline app/models/user.rb:6]: image

File: app/models/user.rb [hunk 3]
```diff
@@ -29,7 +29,7 @@ class User < ApplicationRecord
   after_create :send_account_created_email, if: :persisted?
 
   # Relationships
-  # has_many :call_list_entries # Soon
```
Comment (lomky) [inline app/models/user.rb:32]: image

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -14,7 +14,7 @@
           <%#= link_to 'Sign up', new_registration_path(resource_name), class: 'right' %>
         <% end -%>
         <%= f.email_field :email,
-                          label: t('mongoid.attributes.user.email'),
```
Comment (lomky) [inline app/views/devise/sessions/new.html.erb:17]: No objection here - but is this nonsequitor to the rest of the PR?

File: test/controllers/call_lists_controller_test.rb [hunk 5]
```diff
@@ -65,7 +65,7 @@ class CallListsControllerTest < ActionDispatch::IntegrationTest
       assert_no_difference '@user.call_list_entries.count' do
         patch remove_patient_path(@patient_1), xhr: true
       end
-      assert_response :success
+      assert_response :not_found
```
Comment (lomky) [inline test/controllers/call_lists_controller_test.rb:68]: Why is this flipping?

**Review by lomky — APPROVED** [review]
Couple trivial questions and one possible hanger on - otherwise GTG!

File: app/views/devise/sessions/new.html.erb (hunk 4 above)
Comment (colinxfleming) [reply app/views/devise/sessions/new.html.erb:17]: eeeeeeeeeeh maybe. Sorry, might have gotten turned around on this one on including it, though it is part of the changeset - I'll keep it and be diligent about keeping stuff scoped to models going fwd.

File: test/controllers/call_lists_controller_test.rb (hunk 5 above)
Comment (colinxfleming) [reply test/controllers/call_lists_controller_test.rb:68]: this was a 'how did this ever work' kinda thing. I'm honestly not sure why this catches right in pg but not really in mongo, but a 404 does feel closer to a sensible response here.

**Comment by colinxfleming** [comment]
thank you @lomky !

**Pull request merged**